src/checkers/inference/solver/backend/encoder/AbstractConstraintEncoderFactory.java

```diff
@@ -10,7 +11,7 @@
  *
  * @see ConstraintEncoderFactory
  */
-public abstract class AbstractConstraintEncoderFactory<ConstraintEncodingT> implements ConstraintEncoderFactory<ConstraintEncodingT>{
+public abstract class AbstractConstraintEncoderFactory<FormatTranslatorT, ConstraintEncodingT> implements ConstraintEncoderFactory<ConstraintEncodingT>{
```

Comment:
Shouldn't these type parameters have some bound? Otherwise, what can you do with them?

Reply:
Done. Extends `FormatTranslator`